great_expectations/rule_based_profiler/parameter_builder/multi_batch_parameter_builder.py

```diff
@@ -0,0 +1,100 @@
+from abc import abstractmethod
+from typing import Dict, List, Optional, Union
+
+import great_expectations.exceptions as ge_exceptions
+from great_expectations import DataContext
+from great_expectations.core.batch import BatchRequest
+from great_expectations.rule_based_profiler.domain_builder.domain import Domain
+from great_expectations.rule_based_profiler.parameter_builder.parameter_builder import (
+    ParameterBuilder,
+)
+from great_expectations.rule_based_profiler.parameter_builder.parameter_container import (
+    ParameterContainer,
+)
+from great_expectations.rule_based_profiler.util import (
+    get_batch_ids_from_batch_request,
+    get_batch_ids_from_validator,
+    get_parameter_value_and_validate_return_type,
+)
+from great_expectations.validator.validator import Validator
+
+
+class MultiBatchParameterBuilder(ParameterBuilder):
+    """
+    Defines the abstract MultiBatchParameterBuilder class
+
+    MultiBatchParameterBuilder checks that there are multiple batch ids passed to its "_build_parameters()" method,
+    and uses a configured batch_request parameter to obtain them if they are not.
+
+    This is an abstract class in the sense that instead of implementing the interface method, "_build_parameters()",
```

Comment:
1. I do not see batch_ids as a parameter to _build_parameters(); it's not clear to me whether the docstring or the function is out of sync (or if I'm just missing it); 

2. I think that the batch_request *implies* batch_ids, but they're not otherwise specified; specifically, the batch_ids will be components of the overall domains. So, I think it's a *domain list* that is the operative thing that makes for a MultiBatchParameter builder

Reply:
@jcampbell Why do we need `batch_ids` as a parameter to `_build_parameters()`?  If `_build_parameters()` has a `BatchRequest`, then it can get the validator and batch_ids on its own.  In fact, I am simplifying the code to jut use the batch request to get the validator (and get the loaded batch IDs from there).  But it is not in this branch.

Reply:
I agree with @alexsherstinsky here - I think we make the assumption that the batches for the `MultiBatchParameterBuilder` are all of the batches yielded by the its BatchRequest (and so we don't need to specify batch_ids).